src/test/java/com/cloudbees/plugins/credentials/SecretBytesTest.java

```diff
@@ -81,16 +81,16 @@ public void isSecretBytes() throws Exception {
         byte[] data = text.getBytes();
         String valid = SecretBytes.fromBytes(data).toString();
         assertThat(SecretBytes.isSecretBytes(valid), is(true));
-        assertThat(SecretBytes.isSecretBytes(valid.substring(0, 20)+valid.substring(22)), is(false));
+        String broken = valid.substring(0, 20) + valid.substring(22);
+        assertThat(SecretBytes.isSecretBytes(broken), is(false));
         String invalid =
                 "Fu".equals(valid.substring(20,22))
                         ? valid.substring(0, 20) + "fU" + valid.substring(22)
                         : valid.substring(0, 20) + "Fu" + valid.substring(22);
-        if (SecretBytes.isSecretBytes(invalid)) {
-            assertThat(new String(SecretBytes.fromString(invalid).getPlainData()), not(is(text)));
-        } else {
-            assertThat(SecretBytes.isSecretBytes(invalid), is(false));
-        }
+        assertThat(SecretBytes.isSecretBytes(invalid), is(true));
+        assertThat(new String(SecretBytes.fromString(invalid).getPlainData()), not(is(text)));
```

Comment:
Not sure I follow the previous logic. The `else`-branch was tautological.

Reply:
Always passes for me, but maybe at random the textual substitution will make decryption fail? Will restore the `if`-branch I guess.